### PR TITLE
A0-3625: Migrate fork-off to separate repo

### PR DESCRIPTION
**Pull request by Marcin-Radecki**
This is a companion PR to https://github.com/Cardinal-Cryptography/aleph-node/pull/1565. It is almost the same code that is being removed, except:
* entry in the readme stating so,
* bumping crate version to `2.0.0`. Even though there was no breaking changes in the code, major bump is justified to mark that source code was moved to a different repo
* one workflow was added which builds and checks code on PR

### CODE REVIEW

File: .github/workflows/on-pull-request-commit.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  main:
+    name: Build fork-off
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout git
+        uses: actions/checkout@v4
+
+      - name: Call action get-ref-properties
+        id: get-ref-properties
+        uses: Cardinal-Cryptography/github-actions/get-ref-properties@v6
```
Comment (pmikolajczyk41) [inline .github/workflows/on-pull-request-commit.yml]: is it needed?

File: Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+[package]
+name = "fork-off"
+version = "2.0.0"
+edition = "2021"
+license = "Apache 2.0"
+
```
Comment (pmikolajczyk41) [inline Cargo.toml:11]: nit: I suggest adding more metadata (authors, repository, homepage, description, readme)

**Review by pmikolajczyk41 — APPROVED** [review]

File: .github/workflows/on-pull-request-commit.yml (hunk 1 above)
Comment (Marcin-Radecki) [reply .github/workflows/on-pull-request-commit.yml]: Good point, removed

File: Cargo.toml (hunk 2 above)
Comment (Marcin-Radecki) [reply Cargo.toml:11]: Done

File: .github/workflows/on-pull-request-commit.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        uses: actions/checkout@v4
+
+      - name: Install Rust toolchain
+        uses: Cardinal-Cryptography/github-actions/install-rust-toolchain@v6
+
+      - name: Build
+        run: |
+          cargo build --release
+
+      - name: Check
```
Comment (fbielejec) [inline .github/workflows/on-pull-request-commit.yml]: I'd move things around, i.e. first check then build (as check is faster than build, doesn't generate binaries)

File: .github/workflows/on-pull-request-commit.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+---
```
Comment (fbielejec) [inline .github/workflows/on-pull-request-commit.yml:1]: are we not publishing fork-off anywhere? E.g. we could push the docker image

File: docker/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,12 @@
+FROM ubuntu:jammy-20220531
```
Comment (fbielejec) [inline docker/Dockerfile:1]: is this used anywhere in the pipelines, doesn't seems to be?

File: .github/workflows/on-pull-request-commit.yml (hunk 4 above)
Comment (Marcin-Radecki) [reply .github/workflows/on-pull-request-commit.yml:1]: Good point, aded push to ECR workflow

File: .github/workflows/on-pull-request-commit.yml (hunk 3 above)
Comment (Marcin-Radecki) [reply .github/workflows/on-pull-request-commit.yml]: Done

File: docker/Dockerfile (hunk 5 above)
Comment (Marcin-Radecki) [reply docker/Dockerfile:1]: Now it's used, thanks

**Review by fbielejec — APPROVED** [review]

**Pull request merged**